applications/Tellor.md

```diff
@@ -0,0 +1,231 @@
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License |  MIT|
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial**. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1 | Develop Controller contracts | We will provide a set of solidity smart contracts with the functionality described above  |
+| 2 | Develop Parachain integration contract| We will provide an integration contract |
+| 3 | Local testing| |
+
+
+- **Estimated Duration:** 2 months
+- **FTE:**  2
+- **Costs:** 20,000 USD
+
+Details:  A new Substrate pallet will be required which includes the core oracle data reporting and querying logic as well as staking rewards and payment logic, ported from the existing tellorFlex (oracle), AutoPay and UsingTellor (consumer) smart contracts. All the logic within the pallet will simply use the token of the parachain. Any parachain user can create and fund feeds, which are fulfilled by oracle reporters which have deposited a stake to the Oracle contract on the smart contract parachain.  Once completed, integrating and testing the pallet and the controller contracts will take significant work and documentation in order to make a robust and user-friendly oracle module.  
+
+| Number | Deliverable | Specification                                                                                                                                                   |
+| -----: | ----------- | ------------- |
+| **0a.** | License | MIT                                                                                                                                                             |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial**.                                                                             |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1 | Substrate Oracle pallet design and integration | We will provide the Substrate oracle pallet                                                                                                                     |
+| 2 | Testing the pallet and testing integration with projects and selected parachains| We will test full functionality of the system for interactions between the consumer chain and oracle pallet                                                     |
```

Comment:
What exactly is being delivered here? A test report? There is already a `Testing and Testing Guide` deliverable above, how is this different?

Reply:
Hi @semuelle ,
I think it could be part of deliverable *0c but I just wanted to spell out that we have to test milestone 1 and 2 interoperability, happy to remove it though.